### PR TITLE
docs: document `flox build` in manpages

### PR DESCRIPTION
**Pull request by ysndr**
* Describe manifest defined builds and general build customization
such as sandbox control and files.
Add examples for simple pure builds and staged builds
by the example of a simple node built website.

* Add documentation of the `build` section in the manifest.toml

### CODE REVIEW

**Comment by stahnma** [comment]
Shouldn't this be targeted at preview branch?

**Comment by ysndr** [comment]
Even though manpages end up in floxdocs eventually, there is no preview docs branch for manpages.
Also both the command manpage and the manifest section have are marked as experimental.
I think having at least some documentation _for what works today_ is better than not surfacing it at all.
Regarding tutorials and detailed docs in floxdocs, those might wait or be developed on preview until we know better how this feature is going to be used.

**Comment by mkenigs** [comment]
> Even though manpages end up in floxdocs eventually, there is no preview docs branch for manpages. Also both the command manpage and the manifest section have are marked as experimental. I think having at least some documentation _for what works today_ is better than not surfacing it at all. Regarding tutorials and detailed docs in floxdocs, those might wait or be developed on preview until we know better how this feature is going to be used.

Since we haven't released the feature yet, I think doing what we have for other large features and creating a `doc-build` directory makes sense

File: cli/flox/doc/flox-build.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+     [-d=<path>]
+     [-L]
+     [<package>]...
+```
+
+# DESCRIPTION
+
+Build the specified `<package`> to run or verify your software locally.
+Create build artifacts from instructions in the manifest
+of the environment in `<path>`.
```
Comment (mkenigs) [inline cli/flox/doc/flox-build.md:29]:
```suggestion
Build the specified `<package>` from the environment in `<path>`,
and output build artifacts at `result-<package>` adjacent to the environment.
```

File: cli/flox/doc/flox-build.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+Possible values for `<package>` are all keys under the `build` attribute
+in the `manifest.toml`.
+If no `<package>` is specified, flox will attempt to build all packages
+that are defined in the environment.
+
+Packages are built by running the script defined in `build.<package>.command`
+within a `bash` subshell.
+The shell will behave as if `flox activate` was run
+immidiately prior to running the build script.
```
Comment (mkenigs) [inline cli/flox/doc/flox-build.md:41]:
```suggestion
immediately prior to running the build script.
```

File: cli/flox/doc/flox-build.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+The current working directory of the script will be set to the root
+of the `git` repository in which the environment is defined.
+Consequently, _building requires your environment to be tracked
+under `git` version control_.
```
Comment (mkenigs) [inline cli/flox/doc/flox-build.md:45]: nonblocking: explain why (I don't even know)

File: cli/flox/doc/flox-build.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+By default, builds are run in a sandbox.
+Within the sandbox only the files tracked by git are available.
+Untracked files, files outside of the repository and notably network access
+will be restricted to encourage "pure" and reproducible builds.
+Builds can opt-out of the sandbox by setting `build.<package>.sandbox = "off"`.
+With the sandbox disabled, building is equivalent
+to running the build script manually within a shell created by `flox activate`.
+Any build can access the _results_ of other builds
+(including non-sandboxed ones) by referring to their name via `${<package>}`.
```
Comment (mkenigs) [inline cli/flox/doc/flox-build.md:55]: nonblocking
```suggestion
(including non-sandboxed ones) by referring to their name via `${<package>}`.
In the example below, the `app` package depends on the `dep` package by using `${deps}/node_modules`.
```

File: cli/flox/doc/flox-build.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+To produce build results, the build script is expected
+to copy or move artifacts to a special folder referenced by `$out`.
```
Comment (mkenigs) [inline cli/flox/doc/flox-build.md:58]:
```suggestion
`flox build` creates a temporary directory for the build script to output build artifacts to.
The environment variable `out` is set to this directory,
and the build script is expected to copy or move artifacts to `$out`.
```

File: cli/flox/doc/flox-build.md [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+Alternatively, explicit artifacts can be declared via `build.<package>.files`.
```
Comment (mkenigs) [inline cli/flox/doc/flox-build.md:59]: question blocking: I don't think it's quite clear how this works. Where do those files have to be created?

File: cli/flox/doc/flox-build.md [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+
+```shell
+$ npx serve result-app
+```
+
+# SEE ALSO
+
+[`flox-activate(1)`](./flox-activate.md)
+[`manifest.toml(5)`](./manifest.toml.md)
+[`docker-load(1)`]
```
Comment (mkenigs) [inline cli/flox/doc/flox-build.md:179]:
```suggestion
```

**Review by mkenigs — CHANGES_REQUESTED** [review]

File: cli/flox/doc/flox-build.md (hunk 3 above)
Comment (ysndr) [reply cli/flox/doc/flox-build.md:45]: fwiw i dont think the sentence before that is even truthful currently, ive been buildin in a subdir and _not_ have the cwd set to anything but the dir containing the .flox (which tbh, is the more sensible behavior in my opinion)
The restriction also only concerns sandboxed builds. i'll remove this here and amend something in the paragraph below.

File: cli/flox/doc/flox-build.md (hunk 6 above)
Comment (ysndr) [reply cli/flox/doc/flox-build.md:59]: The files attibute is a bit unclear in general.
I also think there should eb a way to specify where the files will be located in the final package.
To add to this, i'm not even sure if the files attribute is already supported in the first place.
I'll take this out for now.

**Comment by ysndr** [comment]
Also moved the docs to doc-build since that seemed to be preferred

**Review by mkenigs — APPROVED** [review]

**Pull request merged**